jenkins/ci.suse.de/pipelines/openstack-crowbar.Jenkinsfile

```diff
@@ -0,0 +1,295 @@
+/**
+ * The openstack-crowbar Jenkins Pipeline
+ */
+
+def ardana_lib = null
+
+pipeline {
+  options {
+    // skip the default checkout, because we want to use a custom path
+    skipDefaultCheckout()
+    timestamps()
+    // reserve a resource if instructed to do so, otherwise use a dummy resource
+    // and a zero quantity to fool Jenkins into thinking it reserved a resource when in fact it didn't
+    lock(label: reserve_env == 'true' ? ardana_env:'dummy-resource',
+         variable: 'reserved_env',
+         quantity: reserve_env == 'true' ? 1:0 )
+  }
+
+  agent {
+    node {
+      label 'cloud-ardana-ci'
+      customWorkspace "${JOB_NAME}-${BUILD_NUMBER}"
+    }
+  }
+
+  stages {
+    stage('Setup workspace') {
+      steps {
+        script {
+          // Set this variable to be used by upstream builds
+          env.blue_ocean_buildurl = env.RUN_DISPLAY_URL
+          env.cloud_type = "virtual"
+          if (ardana_env == '') {
```

Comment:
`ardana_env` has been declared [empty](https://github.com/SUSE-Cloud/automation/pull/3159/files#diff-0e2e74e3c649e6cd1dec8437cc1dafd1R7). I think it expects some value here.

Reply:
True, `ardana_env` has been declared empty because for now the `openstack-crowbar` job is only triggered manually and the user needs to set this parameter to a unique value. The value is then used to name the heat stack that is instantiated in the engineering cloud and all its resources.

Reply:
There is even a [parameter regex](https://github.com/SUSE-Cloud/automation/pull/3159/files#diff-41458080e7dd3e59c3c5ff64f7b63139R17) that prevents the user from starting a job without setting this parameter to a non-empty value.